apps/hashdotdev/src/_pages/blog/0014_structured-data-value.mdx

```diff
@@ -0,0 +1,109 @@
+
+The role of structured data in all this is simple.
+
+**Predictive simulations of real-world things are extremely easy to get wrong.** Many interesting systems that users typically want to simulate end up being "complex systems" in some respect, demonstrating some degree of nonlinearity due to emergent or adaptive behaviors.
+
+Complex systems are typically extremely sensitive to their initial conditions. Ensuring an as close to accurate starting or "initial state" is therefore key. Pre-structured data, especially if itself tested and reviewed, or produced in accordance with a well-understood process or pipeline, can improve confidence in this.
+
+In addition to providing the initial state of a simulation, structured data can be used directly for verification and validation purposes, including as a means of backtesting simulation experiments, and individual behaviors themselves. In a world where behaviors and whole simulation models may be generated by foundation models, this provides humans with an improved sense of their likely correctness.
+
+## More for humans than machines
+
+All this serves to highlight an interesting possibility. If to date structured data has existed for the benefit of _machines_, is it possible that in the future its primary use will in fact me to us, _humans_, in verifying and validating the outputs of machine-generated models: simulation, language, multimodal, or other?
+
+Inherently tied to the idea of structured data are _types_. Data may be of a certain 'type' if it follows a set of rules established by that type.
+
+One way of thinking about structured data is the [Block Protocol type system](https://blockprotocol.org/docs/types), which outlines three different kinds of types: data types, property types, and entity types.
+
+1. **Entites** (semantic 'things' in a database) have entity types.
+  - **Entity types** describe the _properties_ and type(s) of any outgoing _link entities_ that can be associated with an entity.
+  - **Link entities** are the relationships, or 'things' that connect other entities. For example, one `Person` entity may be connected by a linked entity called `Spouse` to another `Person`.
+  - Right now, entities can only have one entity type, but we intend to support entities with multiple entity types in the specification soon.
+2. **Properties** contain data associated with some particular aspect of an entity.
+  - **Property types** describe the possible values can be associated with a property, by way of reference to _data types_ and other properties.
+3. **Data types** describe a space of possible valid values (e.g. `string`, `number` or `boolean`).
+  - Support for defining "custom" data types, i.e. "non-primitive" data types which are not found in the Block Protocol's [graph module specification](https://blockprotocol.org/docs/spec/graph) is coming soon.
```

Comment:
I wouldn't say 'non-primitive' as they still describe primitive values (e.g. a 'string'), it's just that they are more constrained in what values are valid than the default ones
```suggestion
  - Support for defining "custom" data types which are not found in the Block Protocol's [graph module specification](https://blockprotocol.org/docs/spec/graph) is coming soon (e.g. numbers within a certain range, or strings matching a certain pattern).
```

Reply:
@CiaranMn FYI this was pulled from the RFC PR title, so we may want to change: https://github.com/blockprotocol/blockprotocol/pull/355 cc @thehabbos007 @Alfred-Mountfield

Reply:
Happy to change it too, it was a somewhat temporary title, and was suggested by @benwerner01. 

Previously we were calling them "custom data types" or "user data types" but he suggested 'non-primitive' to avoid some confusion (unfortunately I can't remember what the confusion was), and non-primitive was due to us referring to the base data types _as_ primitive data types within the spec, [see here](https://github.com/blockprotocol/blockprotocol/blob/main/rfcs/text/0352-graph-type-system.md#primitive-top-level-data-types) and [here](https://github.com/blockprotocol/blockprotocol/blob/main/rfcs/text/0352-graph-type-system.md#primitive-top-level-data-types-1).

Reply:
(I've unresolved it just to make this thread more discoverable, not asking for any further changes)

Reply:
User-defined or custom sounds fine to me.